public/app/plugins/datasource/prometheus/datasource.ts

```diff
@@ -294,15 +294,20 @@ export class PrometheusDatasource {
           .value();
 
         for (let value of series.values) {
-          if (value[1] === '1') {
+          if (value[1] === '1' || annotation.seriesValueAsTimestamp) {
```

Comment:
We should improve the readability of the code here.
how about storing `value[1] === '1' in a variable and give it a more descriptive name?